### PR TITLE
update: do not capture during night

### PR DESCRIPTION
**Pull request by Adman**
Signed-off-by: Adrian Matejov a.matejov@centrum.sk

### CODE REVIEW

**Comment by Adman** [comment]
@mrshu @Foxiy Please let me know whether this is a good way of detecting the bright/dark images. I'm kinda newbie to this opencv stuff and will appreciate any good advices.

File: scripts/capture.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import os
+
+OUTPUT_DIR = '/home/pi/data'
+DIR = '%Y-%m-%d'
+FILE = '%H:%M:%S.png'
+
+cap = cv2.VideoCapture(0)
+
+#  warmup camera
+for i in range(15):
```
Comment (mrshu) [inline scripts/capture.py]: @Adman the `15` in here needs to be somehow configurable.

File: scripts/capture.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+ret, frame = cap.read()
+grayimg = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+
+_, thresh = cv2.threshold(grayimg, 80, 255, cv2.THRESH_BINARY)
+_, thresh_inv = cv2.threshold(grayimg, 80, 255, cv2.THRESH_BINARY_INV)
+nonzero = cv2.countNonZero(thresh)
+nonzero_inv = cv2.countNonZero(thresh_inv)
+
+if nonzero - nonzero_inv > 0:
```
Comment (mrshu) [inline scripts/capture.py]: @Foxiy what is your view on this? I am not convinced this is the best approach and believe that we should discuss it further.

**Comment by mrshu** [comment]
@Adman Nice first pass. I think it needs a bit more thought though. Let's sync up on it with @Foxiy in the next few days.

File: scripts/capture.py (hunk 1 above)
Comment (vidriduch) [reply scripts/capture.py]: I would strongly recommend to add or configuration file or (and this is my preferred option) a way tto set it from cli maybe something like `--warmup 15` or `-w 15`

File: scripts/capture.py (hunk 2 above)
Comment (vidriduch) [reply scripts/capture.py]: Well I think we definitely shouldn't do two thresholds. We are effectively asking here how many pixels there are that have lower intensity value that 80 from all the pixels. So same effect can be probably achieved by just comparing it to 1/3 of total number of pixels or by replacing the second threshold with numpy operation something like `np.count_nonzero(grayimg < 80)`. Or you can just simply ask what is the highest intensity pixel with max and if it is less then lets say 20 then everything else is lower therefore even blacker (but something more cleverer :) something like top 10 pixels or something)

File: scripts/capture.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+DIR = '%Y-%m-%d'
+FILE = '%H:%M:%S.png'
+
+cap = cv2.VideoCapture(0)
+
+#  warmup camera
+for i in range(15):
+    _, _ = cap.read()
+
+ret, frame = cap.read()
```
Comment (vidriduch) [inline scripts/capture.py]: @Adman you should always check the return value. It will save you from lot of headaches.

**Review by vidriduch — CHANGES_REQUESTED** [review]

**Comment by Adman** [comment]
@mrshu @Foxiy warmup is now configurable. Also, I got rid of 2 thresholds and now the image is being saved only when there is less than 1/3 of dark pixels.

File: scripts/capture.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+OUTPUT_DIR = '/home/pi/data'
+DIR = '%Y-%m-%d'
+FILE = '%H:%M:%S.png'
+
+
+def capture_and_save(warmup=15):
+    cap = cv2.VideoCapture(0)
+
+    #  warmup camera
+    for i in range(15):
```
Comment (mrshu) [inline scripts/capture.py]: @Adman there is `warmup` now, it seems that it should be used rather than a hardcoded `15`.

File: scripts/capture.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        _, _ = cap.read()
+
+    ret, frame = cap.read()
+    if not ret:
+        print('Failed to capture frame, closing...')
+        return
+
+    grayimg = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+
+    _, thresh = cv2.threshold(grayimg, 60, 255, cv2.THRESH_BINARY)
```
Comment (mrshu) [inline scripts/capture.py]: @Adman It seems a comment here (especially on the `60`) would be in order.

**Review by mrshu — CHANGES_REQUESTED** [review]
@Adman Looks pretty nice, just couple of suggestions.

**Comment by mrshu** [comment]
@Adman looks good and ready for merge from my point of view.

My only concern is that I never saw `requirements.pip` -- it seems that `requirements.txt` is much more common.

**Review by mrshu — APPROVED** [review]

**Comment by vidriduch** [comment]
@Adman :+1:  LGTM. Thanks for making corrections, nicely done. Lets get it merged.

**Review by vidriduch — APPROVED** [review]

**Pull request merged**